Cargo.toml

```diff
@@ -4,4 +4,5 @@ members = [
     "math",
     "crypto",
     "proving-system/stark",
+    "gpu",
```

Comment:
Is it going to be shared between multiple GPUs ? Else I'd call this metal. Or metal GPU. Or it can be like proving-system, where you have a folder with multiple workspaces, one for each proving system

Reply:
The idea is to have multiple platforms in this crate for different gpus. Right now we only have Metal and in the coming days we'll probably have something with CUDA. We may need to reorg a bit like you suggested and have different modules for each platform. @mpaulucci @IAvecilla wdyt?

Reply:
I like the idea of following the proving-system convention and creating a metal dir and declaring `gpu/metal` here. Right now CUDA is in early development so we can change the name to _metal_ if you think is a better approach